File: xv/patch/XV2020Voice.swift

extension XV2020 {
  
  enum Voice {
    
    static let patchWerk = XV.Voice.patchWerk(common: Common.patchWerk, tone: Tone.patchWerk, fx: FX.patchWerk, chorus: Chorus.patchWerk, reverb: Reverb.patchWerk, initFile: "xv5050-voice-init")
    
    static let bankWerk = XV.Voice.bankWerk(patchWerk)
    
    enum Common {
      static let patchWerk = XV.Voice.Common.patchWerk(parms.params())
      
      static let parms = XV5050.Voice.Common.parms + [
        .p([.out, .assign], 0x27, .options(outAssignOptions))
      ]
      
      static let outAssignOptions = Tone.outAssignOptions
        <<< [13 : "Tone"]
    }
    
    enum Tone {
      static let patchWerk = XV.Voice.Tone.patchWerk(params: parms.params())
      
      static let parms = XV5050.Voice.Tone.parms + [
        .p([.out, .assign], 0x11, .options(outAssignOptions))
      ]

      static let waveGroupOptions = XV5050.Voice.Tone.waveGroupOptions

      static let outAssignOptions = [
        0 : "MFX",
        1 : "A",
        5 : "1",
        6 : "2",
      ]
    }
    
  }
}

File: xv/xv2020_voice.js
const XVVoice = require('./xv_voice.js')
const XV5050Voice = require('./xv5050_voice.js')

const outAssignOptions = [
  0 : "MFX",
  1 : "A",
  5 : "1",
  6 : "2",
]

const toneParms = XV5050Voice.toneParms.concat([
  ['out/assign', { b: 0x11, opts: outAssignOptions }]
])

const tonePatchWerk = XVVoice.tonePatchWerk(toneParms)

const commonOutAssignOptions = toneOutAssignOptions
commonOutAssignOptions[13] = "Tone"

const commonParms = XV5050Voice.commonParms.concat([
  ['out/assign', { b: 0x27, opts: commonOutAssignOptions }]
])

const commonPatchWerk = XVVoice.commonPatchWerk(commonParms)


const patchWerk = XVVoice.patchWerk({
  common: commonPatchWerk, 
  tone: Tone.patchWerk, 
  fx: FX.patchWerk,
  chorus: Chorus.patchWerk, 
  reverb: Reverb.patchWerk,
}, "xv5050-voice-init")

const bankWerk = XVVoice.bankWerk(patchWerk)

module.exports = {
  patchWerk,
  bankWerk,
}